test/functional/test_framework/ellsq.py

```diff
@@ -0,0 +1,245 @@
+# Source: https://github.com/sipa/writeups/tree/main/elligator-square-for-bn
+"""Test-only Elligator Squared implementation
+
+WARNING: This code is slow, uses bad randomness, does not properly protect
+keys, and is trivially vulnerable to side channel attacks. Do not use for
```

Comment:
Slow and bad randomness apply, but elligator square inherently doesn't deal with secret keys, so there is nothing to protect, and side channel attacks do not apply.

Reply:
Thanks! Done.